scripts/windows/InstallWindowsAgent.ps1

```diff
@@ -79,11 +34,14 @@ Function DownloadAndInstallAgent(
         # Run Installer
         InstallAgent
 
-    }
-    If (CheckProgramInstalled("Microsoft Visual C\+\+ 2013 x64") -and CheckProgramInstalled("Microsoft Visual C\+\+ 2013 x86") -and AgentIsOnFileSystem) {
-        Write-Output 'JumpCloud Agent Installer Completed'
-    } Else {
-        Write-Output 'JumpCloud Agent Installer Failed'
+        # Check if Agent is on filesystem
+        Start-Sleep -Seconds 10 # Wait 10 seconds for the agent install to complete before checking if the agent is on the filesystem
+
+        If (Test-Path -Path "$($AGENT_PATH)\$($AGENT_BINARY_NAME)") {
```

Comment:
A better check would be to us sc.exe to check if the agent is running as a service.  Sometimes the agent is laid down but something went wrong during installation and the service didn't get registered.  This can be reproduced pretty easily but installing with the wrong connect key.

Reply:
Added a check for the service using Get-Service, equivalent to sc.exe query